presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -2713,6 +2713,11 @@ public void testNonEqualityLeftJoin()
         assertQuery(
                 "SELECT * FROM (VALUES 1) t1(a) LEFT OUTER JOIN (VALUES (1,2,2), (1,2,3), (1, 2, NULL)) t2(x,y,z) ON a=x AND y = z",
                 "VALUES (1, 1, 2, 2)");
+
+        // left join which gets converted to inner join without equality conditions.
+        // all symbols pruned by original join
```

Comment:
It is important that original join did not output symbols needed by filter function (`o.orderkey < l.orderkey` in this example).
This is the failing case.